redash/permissions.py

```diff
@@ -14,24 +14,24 @@
 ACCESS_TYPES = (ACCESS_TYPE_VIEW, ACCESS_TYPE_MODIFY, ACCESS_TYPE_DELETE)
 
 
-def has_access(object_groups, user, need_view_only):
+def has_access(object, user, need_view_only):
```

Comment:
The purpose of switching to passing the `object` instead of `object.groups` is to be able to use a different logic when needed. For example: when using a query API key just assert that it "belongs" to this query.

Only changing where we pass the groups, means we still have the need to set pseudo groups on the API Key object, which is what we wanted to resolve. If I'm not mistaken, in this form we don't prevent query API key from running _any_ parameter.